cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -269,6 +277,37 @@ class device_row_comparator {
   std::optional<device_span<null_order const>> const _null_precedence;
 };  // class device_row_comparator
 
+/**
+ * @brief Wraps and interprets the result of templated Comparator that returns a weak_ordering.
+ * Returns true if the weak_ordering matches any of the templated values.
```

Comment:
Addressing @jrhemstad's request for documentation about when to use an equality comparator instead:
```suggestion
 * Returns true if the weak_ordering matches any of the templated values.
 *
 * Note that this should never be used with only `weak_ordering::EQUIVALENT`.
 * An equality comparator should be used instead for optimal performance.
```